tests/store/api/find/test_find_business_data_sample.py

```diff
@@ -99,6 +99,24 @@ def test__get_sample_by_name(store_with_samples_that_have_names: Store, name="sa
     assert samples and samples.name == name
 
 
+def test_get_samples_by_name_pattern(
+    store_with_samples_that_have_names: Store, name_pattern="sample"
+):
+    """Test that samples can be fetched by name."""
+    # GIVEN a database with two samples of which one has a name
+
+    # ASSERT that there are two samples in the store
+    assert len(store_with_samples_that_have_names.get_samples()) == 4
```

Comment:
Better to fetch all entries and do .count() or len to get the actual number of entries instead of hard coding it.

Reply:
It is a list of sample .count() does not work here